src/octod/api/repos/ReposGitlab.d

```diff
@@ -0,0 +1,259 @@
+/*******************************************************************************
+
+    Library release tool.
+
+    Copyright:
+        Copyright (c) 2018 sociomantic labs GmbH. All rights reserved.
+
+    License:
+        Boost Software License Version 1.0. See LICENSE.txt for details.
+
+*******************************************************************************/
+
+module octod.api.repos.ReposGitlab;
+
+import octod.api.Repos;
+
+/// Gitlab specific implementation of Repository
+class GitlabRepo : Repository
+{
+    import octod.core;
+    import vibe.data.json;
+    import std.typecons;
+
+    /// Full repository identifier
+    protected string repo;
+
+    /// Repo name & login
+    protected string _name, _login;
+
+    /** C'tor
+
+        Params:
+            con = http connection
+            json = initial json info
+            repo = repo name
+
+     **/
+    public this ( HTTPConnection* con, const Json json, string repo )
```

Comment:
So it looks like gitlab doesn't have single "repo" json with all the data and you have to make request for individual methods, correct? In that case there is no point in keepin `Json` field in `Repository` base class - make it specifc to GitHub implementation.